[BE][inductor] replace "and" -> "logical_and" in bucketize_binary_search

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #160941

Get rid of these warnings:
```
/home/dberard/local/pytorch-env7/pytorch/torch/_inductor/runtime/triton_helpers.py:317: UserWarning: Logical operators 'and' and 'or' are deprecated for non-scalar tensors;
 please use '&' or '|' instead
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

davidberard98 committed:
[BE][inductor] replace "and" -> "&" in bucketize_binary_search

[ghstack-poisoned]

malfet commented on torch/_inductor/runtime/triton_helpers.py:
```diff
@@ -314,8 +314,8 @@ def bucketize_binary_search(
-            mid * BOUNDARIES_STRIDE + boundary_indices
-        ) < BOUNDARIES_UNDERLYING_NUMEL and mid < BOUNDARIES_SIZE
+            (mid * BOUNDARIES_STRIDE + boundary_indices) < BOUNDARIES_UNDERLYING_NUMEL
+        ) & (mid < BOUNDARIES_SIZE)
```
IMO this makes code more readable as it's clear that operation is performed on tensors rather than scalars...
```suggestion
        ).logical_and(mid < BOUNDARIES_SIZE)
```

malfet reviewed (approved):
I have to say, it feels a bit weird, may be one should revisit the deprecation message? I.e. `and` is very idiomatic Python and `&` feels like a wrong operator to use to me (as it's a bitwise and rather than logical_and)

davidberard98 committed:
Update on "[BE][inductor] replace "and" -> "&" in bucketize_binary_search"


Get rid of these warnings:
```
/home/dberard/local/pytorch-env7/pytorch/torch/_inductor/runtime/triton_helpers.py:317: UserWarning: Logical operators 'and' and 'or' are deprecated for non-scalar tensors;
 please use '&' or '|' instead
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

davidberard98 committed:
Update on "[BE][inductor] replace "and" -> "logical_and" in bucketize_binary_search"


Get rid of these warnings:
```
/home/dberard/local/pytorch-env7/pytorch/torch/_inductor/runtime/triton_helpers.py:317: UserWarning: Logical operators 'and' and 'or' are deprecated for non-scalar tensors;
 please use '&' or '|' instead
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]
